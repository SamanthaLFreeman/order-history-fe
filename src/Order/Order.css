.order {
  background-color: #fff;
  box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.3), 0 6px 20px 0 rgba(0, 0, 0, 0.3);
  display: flex;
  margin: 30px 0;
  width: 80vw;
}

img {
  margin: 8px;
  width: 200px;
}

.order-center {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 15px;
  width: 60%;
}

.order-right {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  justify-content: space-between;
  margin-top: 15px;
}

.price {
  font-size: 1.2em;
  font-weight: bold;
}

.remove-btn {
  background-color: #9c0000;
  font-size: 1em;
  font-weight: normal;
  height: 90px;
  width: 200px;
}
