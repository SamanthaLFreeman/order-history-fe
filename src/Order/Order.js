import React from 'react';
import './Order.css';

const Order = ({order}) => {
  return (
    <h2>{order.name}</h2>
  )
}

export default Order;