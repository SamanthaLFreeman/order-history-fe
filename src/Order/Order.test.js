import React from 'react';
import { shallow } from 'enzyme';
import Order from './Order';

describe('Order', () => {
  let wrapper;
  let removeMock;
  
  beforeEach(() => {
    const mockOrder = {
      name: 'Jeans',
      img: 'fancy looking jeans',
      description: 'Obviously pants...',
      price: '10000000'
    }
    removeMock = jest.fn()
    wrapper = shallow(<Order order={mockOrder} removeOrder={removeMock}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call the removeOrder method on click of button', () => {
    wrapper.find('button').simulate('click');
    expect(removeMock).toHaveBeenCalled();
  })
})