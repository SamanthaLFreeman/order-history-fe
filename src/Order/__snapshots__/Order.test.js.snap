// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot 1`] = `
<section>
  <img
    alt="Jeans"
    src="fancy looking jeans"
  />
  <h2>
    Jeans
  </h2>
  <p>
    Obviously pants...
  </p>
  <p>
    $
    10000000
  </p>
  <button
    onClick={[MockFunction]}
  >
    Remove From History
  </button>
</section>
`;
